src/main/java/bio/terra/service/filedata/azure/AzureSynapsePdao.java

```diff
@@ -0,0 +1,248 @@
+package bio.terra.service.filedata.azure;
+
+import bio.terra.common.Column;
+import bio.terra.common.SynapseColumn;
+import bio.terra.model.BillingProfileModel;
+import bio.terra.model.IngestRequestModel.FormatEnum;
+import bio.terra.service.dataset.DatasetTable;
+import bio.terra.service.filedata.azure.blobstore.AzureBlobStorePdao;
+import bio.terra.service.filedata.azure.util.BlobContainerClientFactory;
+import bio.terra.service.filedata.azure.util.BlobSasTokenOptions;
+import bio.terra.service.resourcemanagement.azure.AzureResourceConfiguration;
+import bio.terra.service.resourcemanagement.azure.AzureStorageAccountResource;
+import bio.terra.service.resourcemanagement.azure.AzureStorageAccountResource.ContainerType;
+import com.azure.core.credential.AzureSasCredential;
+import com.azure.storage.blob.BlobUrlParts;
+import com.azure.storage.blob.sas.BlobSasPermission;
+import com.google.common.annotations.VisibleForTesting;
+import com.microsoft.sqlserver.jdbc.SQLServerDataSource;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.time.Duration;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.NotImplementedException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.stringtemplate.v4.ST;
+
+@Component
+public class AzureSynapsePdao {
+  private static final Logger logger = LoggerFactory.getLogger(AzureSynapsePdao.class);
+
+  private final AzureResourceConfiguration azureResourceConfiguration;
+  private final AzureBlobStorePdao azureBlobStorePdao;
+  private static final String PARSER_VERSION = "2.0";
+  private static final Duration DEFAULT_SAS_TOKEN_EXPIRATION = Duration.ofHours(24);
+  private static final String scopedCredentialCreateTemplate =
+      "CREATE DATABASE SCOPED CREDENTIAL [<scopedCredentialName>]\n"
+          + "WITH IDENTITY = 'SHARED ACCESS SIGNATURE',\n"
+          + "SECRET = '<secret>';";
+
+  private static final String dataSourceCreateTemplate =
+      "CREATE EXTERNAL DATA SOURCE [<dataSourceName>]\n"
+          + "WITH (\n"
+          + "    LOCATION = '<scheme>://<host>/<container>',\n"
+          + "    CREDENTIAL = [<credential>]\n"
+          + ");";
+
+  private static final String createTableTemplate =
+      "CREATE EXTERNAL TABLE [<tableName>]\n"
```

Comment:
we can omit the newline from the ends of lines here.

Reply:
I think the newlines help with readability of the query when it's printed out. I tested without the newlines and it doesn't seem to space it out correctly without the newline notations.

Reply:
Man, I can't wait for text blocks to be added! (Java 13)